src/webhook/mutation/pod_mutator/oneagent_mutation/versionlabel.go

```diff
@@ -18,11 +26,26 @@ func newVersionLabelMapping(namespace corev1.Namespace) VersionLabelMapping {
 }
 
 func getMappingFromNamespace(namespace corev1.Namespace) VersionLabelMapping {
-	// TODO: Implementation => Parse Namespace annotations into correct format
-	return VersionLabelMapping{}
+	annotationLabelMap := map[string]string{
+		versionMappingAnnotationName: releaseVersionEnv,
+		productMappingAnnotationName: releaseProductEnv,
+		stageMappingAnnotationName:   releaseStageEnv,
+		buildVersionAnnotationName:   releaseBuildVersionEnv,
+	}
+
+	versionLabelMapping := VersionLabelMapping{}
+	for annotationKey, labelKey := range annotationLabelMap {
+		if fieldRef, ok := namespace.Annotations[annotationKey]; ok {
+			versionLabelMapping[labelKey] = fieldRef
+		}
+	}
+
+	return versionLabelMapping
 }
 
 func mergeMappingWithDefault(labelMapping VersionLabelMapping) VersionLabelMapping {
-	// TODO: Implementation => Combine the maps
-	return defaultVersionLabelMapping
+	result := VersionLabelMapping{}
+	maps.Copy(result, defaultVersionLabelMapping)
+	maps.Copy(result, labelMapping)
```

Comment:
does it makes sense to introduce the experimental library, which promises 0 stability to avoid minimal code like
```
func (mapping VersionLabelMapping) Copy(source VersionLabelMapping) {
	for key, value := range source {
		mapping[key] = value
	}
}
```

then the function can be:
```suggestion
	result.Copy(defaultVersionLabelMapping)
	result.Copy(labelMapping)
```

we should move to these fancy functions when they are no longer experimental, but in the meanwhile I feel its less of a headache to just not use the experimental stuff